source/patterns/@aws-solutions-constructs/aws-route53-apigateway/README.md

```diff
@@ -0,0 +1,122 @@
+
+const lambdaProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_12_X,
+    handler: 'index.handler'
+};
+
+const lambdaFunction = new lambda.Function(this, 'LambdaFunction', lambdaProps);
+
+const apiGatewayProps: api.LambdaRestApiProps = {
+    handler: lambdaFunction,
+    endpointConfiguration: {
+        types: [api.EndpointType.REGIONAL]
+    },
+    defaultMethodOptions: {
+        authorizationType: api.AuthorizationType.NONE
+    }
+};
+
+const apiGateway = new api.LambdaRestApi(this, 'LambdaRestApi', apiGatewayProps);
+
+const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
+  zoneName: 'example.com',
+});
+
+// This construct can only be attached to a configured API Gateway.
+new Route53ToApigateway(this, 'Route53ToApigatewayPattern', {
+    existingApiGatewayObj: apiGateway,
+    existingHostedZoneInterface: parentZone,
+    publicApi: true
+});
+
+```
+
+## Initializer
+
+``` text
+new Route53ToApigateway(scope: Construct, id: string, props: Route53ToApigatewayProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`Route53ToApigatewayProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+This construct cannot create a new Public Hosted Zone, if you are creating a public API you must supply an existing Public Hosted Zone that will be reconfigured with a new Alias record. Public Hosted Zones are configured with public domain names and are not well suited to be launched and torn down dynamically, so this construct will only reconfigure existing Public Hosted Zones.
+
+This construct can create Private Hosted Zones. If you want a Private Hosted Zone, then you can either provide an existing Private Hosted Zone or a privateHostedZoneProps value with at least the Domain Name defined.
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+| privateHostedZoneProps? | [route53.PrivateHostedZoneProps](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.PrivateHostedZoneProps.html) | Optional custom properties for a new Private Hosted Zone. Cannot be specified for a public API. Cannot specify a VPC, it will use the VPC in existingVpc or the VPC created by the construct. Providing both this and existingHostedZoneInterface is an error. |
+| existingHostedZoneInterface? | [route53.IHostedZone](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.IHostedZone.html) | Existing Public or Private Hosted Zone (type must match publicApi setting). Specifying both this and privateHostedZoneProps is an error. If this is a Private Hosted Zone, the associated VPC must be provided as the existingVpc property.|
+| publicApi | boolean | Whether the construct is deploying a private or public API. This has implications for the Hosted Zone and VPC. |
```

Comment:
Let's check deeper into this - can our current API Gateway constructs create an internal API Gateway? I'm a little murky in this area - but since none of them involve a VPC I'm pretty sure they are public only. If so, perhaps we need to say this must be true at this time. And perhaps we don't expose privateHostedZoneProps at this time as well.

It all depends on what is a non-public RestAPI from API Gateway and whether our constructs can create one.

OTOH - do we want to limit ourselves to RestAPIs that we create in constructs. I guess it's possible that someone could create a private RestAPI themselves and want to use this. In that case we could just discuss the issue in the docs.